warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/TermFrequencyHitFunction.java

```diff
@@ -0,0 +1,331 @@
+                    String valueField = value + '\u0000' + field;
+                    functionSearchSpace.put(function, valueField);
+                }
+            }
+        }
+    }
+    
+    /**
+     * Get this function's search space as defined by fields and values
+     *
+     * @param function
+     *            a {@link Function} that is either adjacent, phrase, or within
+     * @return a list of fields and values associated with the provided function
+     */
+    private FunctionArgs parseFunction(Function function) {
+        // functions may take different forms..
+        // within = {field, number, termOffsetMap, terms...}
+        // within = {number, termOffsetMap, terms...}
+        // adjacent = {field, termOffsetMap, terms...}
+        // adjacent = {termOffsetMap, terms...}
+        // phrase = {field, termOffsetMap, terms...}
+        // phrase = {termOffsetMap, terms...}
+        List<JexlNode> args = function.args();
+        int index = function.name().equals("within") ? 3 : 2;
+        
+        // If the first arg is a number or termOffsetMap then no field was provided as part of the function.
+        // The fields will be build via value lookups from the tfFVs.
+        boolean specialCase = false;
+        JexlNode first = args.get(0);
+        if (isFirstNodeSpecial(first)) {
+            specialCase = true;
+            index--;
+        }
+        
+        // Parse the values first. Might have to lookup fields by value.
+        List<String> values = getValuesFromArgs(args, index);
+        
+        List<String> fields;
+        if (specialCase) {
+            // field(s) were not present in the function, lookup fields by value.
+            fields = lookupFieldsByValues(values);
+        } else {
+            // function's first arg is the field, or fields in the form (FIELD_A || FIELD_B)
+            fields = parseField(args.get(0));
+        }
+        
+        return new FunctionArgs(fields, values);
+    }
+    
+    // If a function does not contain the fields being queried, find the fields via the term frequency field-value map
+    private List<String> lookupFieldsByValues(List<String> values) {
+        List<String> fields = new LinkedList<>();
+        for (String key : tfFVs.keySet()) {
+            for (String value : values) {
+                if (tfFVs.containsEntry(key, value)) {
+                    fields.add(key);
+                }
+            }
+        }
+        return fields;
+    }
+    
+    // A node is special if it is not the fields being searched. That is, the first node is a number or a variable 'termOffsetMap'
+    private boolean isFirstNodeSpecial(JexlNode node) {
+        if (node instanceof ASTNumberLiteral) {
+            return true;
+        } else if (node instanceof ASTReference) {
+            List<ASTIdentifier> ids = JexlASTHelper.getIdentifiers(node);
+            if (ids.size() == 1 && ids.get(0).image.equals("termOffsetMap")) {
```

Comment:
This can be simplified to `return ids.size() == 1 && ids.get(0).image.equals("termOffsetMap");`